### PR TITLE
Added HelpDesk hierarchy

### PR DESCRIPTION
**Pull request by adudekyeti**
Added HelpDesk hierarchy

### CODE REVIEW

File: modules/HelpDesk/views/Hierarchy.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * Class to show hierarchy.
+ *
```
Comment (mariuszkrzaczkowski) [inline modules/HelpDesk/views/Hierarchy.php]: @package category

File: modules/HelpDesk/models/Module.php [hunk 2]
```diff
@@ -29,4 +29,276 @@ public function getQueryByModuleField($sourceModule, $field, $record, \App\Query
...
+	 * @param int    $recordId
+	 * @param string $recordsType
+	 * @param string $status
+	 *
+	 * @return bool
+	 */
+	public function massUpdateStatus(int $recordId, string $recordsType, string $status): bool
+	{
+		$recordsId = $this->getHierarchyIds($recordId, $recordsType);
+		foreach ($recordsId as $recordId) {
```
Comment (mariuszkrzaczkowski) [inline modules/HelpDesk/models/Module.php]:
```suggestion
		foreach ($this->getHierarchyIds($recordId, $recordsType) as $recordId) {
```

File: modules/HelpDesk/ConfigTemplate.php [hunk 3]
```diff
@@ -37,5 +37,31 @@
 		'description' => 'Check email opt-out',
 		'validation' => '\App\Validator::bool',
 		'sanitization' => '\App\Purifier::bool'
-	]
+	],
+	'COLUMNS_IN_HIERARCHY' => [
```
Comment (rskrzypczak) [inline modules/HelpDesk/ConfigTemplate.php]: camelCase

File: modules/HelpDesk/actions/ChangeStatus.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	/**
+	 * Function to check permission.
+	 *
+	 * @param \App\Request $request
+	 *
+	 * @throws \App\Exceptions\NoPermitted
+	 */
+	public function checkPermission(App\Request $request)
+	{
+		if (!\App\Privilege::isPermitted($request->getModule(), 'EditView')) {
```
Comment (mariuszkrzaczkowski) [inline modules/HelpDesk/actions/ChangeStatus.php]: $request->getInteger('record')

File: modules/HelpDesk/views/Hierarchy.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	 */
+	public function process(App\Request $request)
+	{
+		$viewer = $this->getViewer($request);
+		$moduleName = $request->getModule();
+		$recordModel = Vtiger_Record_Model::getInstanceById($request->getInteger('record'), $moduleName);
+		$hierarchy = $recordModel->getHierarchyDetails();
+		$viewer->assign('MODULE', $moduleName);
+		$viewer->assign('STATUS_PICKLIST', $recordModel->getField('ticketstatus')->getPickListValues());
+		$viewer->assign('HIERARCHY', $hierarchy);
```
Comment (mariuszkrzaczkowski) [inline modules/HelpDesk/views/Hierarchy.php]: $recordModel->getHierarchyDetails()

**Pull request merged**